test/unit/utils/UWebSocketTransport.spec.ts

```diff
@@ -10,270 +15,693 @@ describe('UWebSocketTransport', () => {
-      await uninitializedTransport.sendMessage(message)
-      expect(errorListener).toHaveBeenCalledWith(new Error('Transport is not ready or socket is not connected'))
+  describe('Queue Processing Edge Cases', () => {
+    beforeEach(() => {
+      jest.useFakeTimers({ advanceTimers: true })
     })
 
-    it('should handle cleanup during active message processing', async () => {
-      jest.useFakeTimers()
-      mockSocket.send.mockReturnValue(0) // Force send failure
+    afterEach(() => {
+      jest.useRealTimers()
+    })
 
+    // Coverage for lines 200-208
+    it('should handle message queue state transitions', async () => {
       const message = new Uint8Array([1])
-      const sendPromise = transport.sendMessage(message)
+      mockSocket.send.mockReturnValue(UWebSocketSendResult.DROPPED)
+
+      // Fill queue to near limit
+      const promises = Array(2)
+        .fill(null)
+        .map(() => transport.sendMessage(message))
+
+      // Add catch to prevent unhandled rejections
+      promises.forEach((p) => (p as unknown as Promise<void>)?.catch(() => {}))
+
+      // Process queue
+      await jest.advanceTimersByTimeAsync(0)
+
+      // Simulate transport becoming inactive during processing
       transport.close()
 
-      await expect(sendPromise).rejects.toThrow('Failed to send message: Socket closed')
-      jest.useRealTimers()
+      // Wait for cleanup to complete
+      await jest.runAllTimersAsync()
+
+      await Promise.all(promises.map((p) => expect(p).rejects.toThrow('Connection closed')))
     })
 
-    it('should handle multiple close calls', () => {
+    // Coverage for lines 300-301
+    it('should handle cleanup with pending timeouts', async () => {
+      const message = new Uint8Array([1])
+      mockSocket.send.mockReturnValue(UWebSocketSendResult.DROPPED)
+
+      const sendPromise = transport.sendMessage(message)
+      // Add catch to prevent unhandled rejection
+      ;(sendPromise as unknown as Promise<void>)?.catch(() => {})
+
+      // Let the first attempt process
+      await jest.advanceTimersByTimeAsync(0)
+
+      // Close before timeout triggers
       transport.close()
-      mockSocket.end.mockClear()
-      mockSocket.getUserData.mockReturnValue({ isConnected: false }) // Indicate connection is closed
-      transport.close() // Second close should not call end again
-      expect(mockSocket.end).not.toHaveBeenCalled()
+
+      // Wait for cleanup to complete
+      await jest.runAllTimersAsync()
+
+      await expect(sendPromise).rejects.toThrow('Connection closed')
+
+      // Ensure no more processing occurs
+      await jest.advanceTimersByTimeAsync(5000)
+      expect(mockSocket.send).toHaveBeenCalledTimes(1)
     })
   })
 
-  describe('Error Scenarios', () => {
-    it('should handle socket errors during queue processing', async () => {
+  describe('Transport Error Handling', () => {
+    it('should handle errors during message processing (send error)', async () => {
```

Comment:
Isn't his being already tested on [the lines 119](https://github.com/decentraland/social-service-ea/pull/91/files#diff-b567943d48312e33460faf772f889ffb94a09c02003ba44388e7ea94d9bd02a6R385) and below?